### PR TITLE
Adding cloud snap zombie replicas validation test

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Automating a customer use case where take a lot of local snapshots and deletion
**Which issue(s) this PR fixes** (optional)
Closes #PTX-17088

**Special notes for your reviewer**:
https://aetos.pwx.purestorage.com/resultSet/testSetID/243395

### CODE REVIEW

File: tests/basic/customer_issues_test.go [hunk 1]
```diff
@@ -384,3 +388,186 @@ var _ = Describe("{FordRunFlatResync}", func() {
...
+						log.FailOnError(err, fmt.Sprintf("error running command [%s] on node [%s]", cmd, n.Name))
+						dirContents := strings.Split(output, "\n")
+						postNodeContentsMap[n.Name] = append(postNodeContentsMap[n.Name], dirContents...)
+					}
+					if len(nodeContentsMap[n.Name]) != len(postNodeContentsMap[n.Name]) {
+						dash.VerifySafely(len(postNodeContentsMap[n.Name]), len(nodeContentsMap[n.Name]), fmt.Sprintf("replicas not deleted in node [%s], Actual: [%v], Expected: [%v]", n.Name, postNodeContentsMap[n.Name], nodeContentsMap[n.Name]))
+					}
+				}
+
+				log.Infof("")
```
Comment (vprabhakar-px) [inline tests/basic/customer_issues_test.go:563]: do we need this ?

**Review by vprabhakar-px — APPROVED** [review]
minor NITs

File: drivers/node/ssh/ssh.go [hunk 2]
```diff
@@ -662,6 +662,13 @@ func (s *SSH) doCmdSSH(n node.Node, options node.ConnectionOpts, cmd string, ign
 	}
 	defer session.Close()
 
+	//this is required to run runc commands
+	if strings.Contains(cmd, "runc exec") {
+		// Disable PTY allocation
+		session.RequestPty("dummy", 80, 40, ssh_pkg.TerminalModes{})
+
```
Comment (mkoppal-px) [inline drivers/node/ssh/ssh.go:669]: nit: Delete this empty line

File: tests/basic/storage_pool_test.go [hunk 3]
```diff
@@ -203,7 +202,7 @@ var _ = Describe("{StoragePoolExpandDiskAdd}", func() {
 		})
 
 		Step("Ensure that new pool has been expanded to the expected size", func() {
-			ValidateApplications(contexts)
+			//ValidateApplications(contexts)
```
Comment (mkoppal-px) [inline tests/basic/storage_pool_test.go:205]: Remove it if not required

**Review by mkoppal-px — APPROVED** [review]
LGTM except for minor comments

File: tests/basic/customer_issues_test.go (hunk 1 above)
Comment (lsrinivas-pure) [reply tests/basic/customer_issues_test.go]: fixed

File: drivers/node/ssh/ssh.go [hunk 4]
```diff
@@ -662,6 +662,13 @@ func (s *SSH) doCmdSSH(n node.Node, options node.ConnectionOpts, cmd string, ign
 	}
 	defer session.Close()
 
+	//this is required to run runc commands
```
[the remark this answers is not in the document]
Comment (lsrinivas-pure) [reply drivers/node/ssh/ssh.go]: fixed

File: drivers/node/ssh/ssh.go (hunk 2 above)
Comment (lsrinivas-pure) [reply drivers/node/ssh/ssh.go]: fixed

File: tests/basic/storage_pool_test.go (hunk 3 above)
Comment (lsrinivas-pure) [reply tests/basic/storage_pool_test.go]: fixed

**Pull request merged**